electrumsv/wallet.py

```diff
@@ -242,6 +246,34 @@ def __init__(self, wallet: 'Wallet', row: AccountRow,
         # TODO(rt12) BACKLOG formalise porting/storage of invoice data extracted from storage data
         self.invoices = InvoiceStore({})
 
+    def get_key_txos(self, tx_id: str, keyinstance_id: int) -> Optional[Set[int]]:
+        """get the set of all output indices in a given transaction for a given keyinstance id"""
+        script, _script_bytes, _object = self._get_cached_script(keyinstance_id)
+        if self._key_txos.get(tx_id) is not None:
+            try:
+                return self._key_txos[tx_id][script]
+            except KeyError as e:
+                return
+
+    def add_tx_to_key_txos(self, tx_id: str, tx: Transaction) -> None:
+        # { txids -> { scriptpubkeys: [ <set of txo indices> ]} }
+        if self._key_txos.get(tx_id) is not None:
+            return
+
+        self._key_txos[tx_id] = {}
+        for index, output in enumerate(tx.outputs):
+            output_type = classify_tx_output(output)
```

Comment:
This function does not work. Any solution cannot be based on it.

Reply:
Oh okay. Until yesterday I wasn't even bothering to filter away the OP_RETURN... data carrier txs... and still worked very fast.

Any idea the best way to identify these ones (only reason is that it will needlessly bloat the cache with useless scriptpubkeys of data carriers.)

Reply:
Would you accept parsing the first two bytes of the scriptpubkey 'manually' and checking to see if it is == OP_FALSE OP_RETURN? I mean we both know it would work how we want it to at least?

Reply:
We know what keys we have and what scripts are in use, and in what transactions and outputs they are used, because the incoming key history points it out to us. It seems there are two paths forward, either we create a new script parsing which is an imperfect cache but not necessarily bad, or we work with the key instance system.

Reply:
What do you mean by "work with the key instance system"?

It's easy to go from keyinstance_id -> scriptpubkey (just derive it/you made a cache for it last week). but afaict much harder to go from scriptpubkey -> keyinstance_id (without some messy trial-and-error, try loads of keyinstance_ids to see if it matches approach...

You'll have to elaborate for me to know what you mean

Reply:
I guess converting scriptpubkey -> scripthash would work universally so that's another option to prevent memory bloat?